eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOslice.java

```diff
@@ -50,19 +50,44 @@ public class EOstring$EOslice extends PhDefault {
             final int start = new Param(rho, "start").strong(Long.class).intValue();
             final int length = new Param(rho, "len").strong(Long.class).intValue();
             final int end = length + start;
-            if (start < 0 || start > end || end > str.length()) {
+            final String error = validate(start, end, str.length());
```

Comment:
@mximp why can't you just do this:

```
if (...) {
  throw
}
if (...) {
  throw
}
if (...) {
  throw
}
```

what's the point of increasing the complexity of the code?

Reply:
It would make me to copy "throw" (which in this case is creation of EOError with corresponding msg) 3 times. It's quite verbose and makes main method (constructor) complex and duplicative. Besides parameters validation is quite self-contained logic. Hence I decided to create separate method for this just to increase readability.